### PR TITLE
fix(idd): mark waiverEvidence required in the pre-merge-readiness schema

### LINKED ISSUE #1087: fix(idd): mark waiverEvidence required in the pre-merge-readiness schema
## Background

Parent roadmap: #1076

`schemas/pre-merge-readiness.schema.json` declares `waiverEvidence` as a
property but does not list it in the root `required` array. The CLI emits it
**unconditionally** (`buildPreMergeReadinessSummary` always computes
`waiverEvidence` via `summarizeExternalCheckWaivers`, which always returns all
buckets, and always attaches it to the summary). By contrast `dispositionEvidence`
is genuinely optional (gated on `includeDispositionEvidence`). So consumers must
currently branch on `waiverEvidence` presence even though it is always there.

## Proposed change

Add `"waiverEvidence"` to the root `required` array of
`schemas/pre-merge-readiness.schema.json` so the envelope is always present and
consumers can parse it unconditionally. Leave `dispositionEvidence` optional.

## Acceptance criteria

- `waiverEvidence` is in the schema's root `required`.
- A pre-merge-readiness output without `waiverEvidence` fails validation; a
  normal output (always carrying it) passes.
- `dispositionEvidence` remains optional.
- `pnpm test` (incl. the schema-type reconciliation suite, which already lists
  `waiverEvidence` in `preMergeReadinessKeys`) and `pnpm run lint:minimum` pass.

## Candidate files

- `schemas/pre-merge-readiness.schema.json` (root `required`)
- `tests/schema-validation.test.mts`, `tests/pre-merge-readiness.test.mts`

---

_Autopilot suitability: 5 / 5 -- higher is more autopilot-suitable; below the configured floor is human-oriented._

_Effort: S -- author-estimated size; a soft autopilot selection tie-breaker only._

### PR DESCRIPTION
**Pull request by kurone-kito**
## Summary

`buildPreMergeReadinessSummary` attaches `waiverEvidence` to the summary
**unconditionally** (the summary literal always sets it via
`summarizeExternalCheckWaivers`, which always returns all buckets). Only
`dispositionEvidence` is gated, on `includeDispositionEvidence`. The schema,
however, listed `waiverEvidence` as a property but **not** in the root
`required` array, so consumers had to branch on its presence even though it is
always there.

This marks `waiverEvidence` required so consumers can parse the envelope
unconditionally, while leaving `dispositionEvidence` optional.

## Changes

- `schemas/pre-merge-readiness.schema.json` — add `"waiverEvidence"` to the
  root `required` array.
- `tests/pre-merge-readiness.test.mts` — add a regression test: a normal output
  always carries `waiverEvidence` and validates; an output without it fails
  validation; `dispositionEvidence` stays optional.

## Verification

- `pnpm test` (incl. the schema-type reconciliation suite, which already lists
  `waiverEvidence` in `preMergeReadinessKeys`) — green.
- `pnpm run lint:minimum` — green.

Closes #1087

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated validation rules so a required waiver evidence section must be present in pre-merge readiness documents.
  * Improved checks to ensure summaries still validate with the expected default content and optional evidence fields remain optional.

### CODE REVIEW

File: tests/pre-merge-readiness.test.mts [hunk 1]
```diff
@@ -95,6 +95,37 @@ test('pre-merge readiness optionally emits disposition evidence', () => {
   assert.deepEqual(validate(summary, readinessSchema), []);
 });
 
+test('pre-merge readiness always carries waiverEvidence and the schema requires it', () => {
+  const fixture = readJson('fixtures/pre-merge-readiness/clean.json');
+  const summary = buildPreMergeReadinessSummary(fixture.input, fixture.options);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply tests/pre-merge-readiness.test.mts]: **Accepted** — applied in 0518204: pass includeDispositionEvidence: false explicitly so the no-dispositionEvidence intent does not depend on the fixture default options.

File: tests/pre-merge-readiness.test.mts [hunk 2]
```diff
@@ -95,6 +95,37 @@ test('pre-merge readiness optionally emits disposition evidence', () => {
...
+
+  // The summary literal always attaches waiverEvidence (unlike the gated
+  // dispositionEvidence), so a normal output carries it and validates.
+  assert.ok(
+    summary.waiverEvidence && typeof summary.waiverEvidence === 'object',
+    'waiverEvidence is always present',
+  );
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply tests/pre-merge-readiness.test.mts:110]: **Accepted** — applied in 0518204: switched the waiverEvidence presence check to Object.hasOwn (matching the existing waiverEvidence test) to avoid falsy/inherited edge cases.

File: tests/pre-merge-readiness.test.mts [hunk 3]
```diff
@@ -95,6 +95,37 @@ test('pre-merge readiness optionally emits disposition evidence', () => {
...
+  assert.equal('dispositionEvidence' in summary, false);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply tests/pre-merge-readiness.test.mts]: **Accepted** — applied in 0518204: switched the dispositionEvidence-absence check to Object.hasOwn so it no longer consults the prototype chain.

File: tests/pre-merge-readiness.test.mts [hunk 4]
```diff
@@ -95,6 +95,37 @@ test('pre-merge readiness optionally emits disposition evidence', () => {
...
+  // dispositionEvidence stays optional: an output that never emits it still
+  // validates (the clean summary above), and dropping it from an output that
+  // did emit it also validates.
+  const withDisposition = buildPreMergeReadinessSummary(fixture.input, {
+    ...fixture.options,
+    includeDispositionEvidence: true,
+  });
+  const withoutDisposition = JSON.parse(JSON.stringify(withDisposition));
+  delete withoutDisposition.dispositionEvidence;
+  assert.deepEqual(validate(withoutDisposition, readinessSchema), []);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply tests/pre-merge-readiness.test.mts:133]: **Accepted** — applied in 0518204: now assert dispositionEvidence was emitted under includeDispositionEvidence: true before deleting it, so the optional-field branch fails if the field stops being emitted.

**Comment by kurone-kito** [comment]
advisory-wait-recovery: claude-e5b4c642 051820475578cb241861feab586620a4d072564d 2026-06-26T11:11:53Z

**Comment by kurone-kito** [comment]
_claude-e5b4c642: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
**Accepted** — coderabbitai[bot] reviewed HEAD 051820475578cb241861feab586620a4d072564d and generated no new comments; the summary walkthrough confirms the change. All prior CodeRabbit/Copilot review threads were addressed in 0518204 and resolved. No action required.

**Comment by kurone-kito** [comment]
_claude-e5b4c642: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
**F4 Cleanup Evidence** (session claude-e5b4c642): minimized 1 straggler IDD operational marker (`advisory-wait-recovery`) the post-merge-cleanup workflow scan missed. The other 12 candidates were already minimized by that workflow (see its evidence comment, applied:7). status=applied, failed=0, skipped=12, viewer-cannot-minimize=0.
